jdaviz/app.py

```diff
@@ -1312,51 +1325,98 @@ def remove(stack_items):
-        replace = event.get('replace', False)
+        data_label = self._get_data_item_by_id(event['item_id'])['name']
+        self.remove_data_from_viewer(event['id'], data_label)
 
-        if viewer_item is None:
-            raise ValueError(f'viewer {viewer_id} not found')
+    def vue_data_item_visibility(self, event):
+        self.set_data_visibility(event['id'],
+                                 self._get_data_item_by_id(event['item_id'])['name'],
+                                 visible=event['visible'], replace=event.get('replace', False))
 
-        if replace:
-            selected_items = {item_id: 'visible'}
-        elif checked:
-            selected_items = {**viewer_item['selected_data_items'], item_id: 'visible'}
-        else:
-            selected_items = {k: v for k, v in viewer_item['selected_data_items'].items() if k != item_id}  # noqa
+    def set_data_visibility(self, viewer_reference, data_label, visible=True, replace=False):
+        """
+        Set the visibility of the layers corresponding to ``data_label`` in a given viewer.
 
-        self._update_selected_data_items(viewer_id, selected_items)
+        Parameters
+        ----------
+        viewer_reference : str
+            Reference (or ID) of the viewer
+        data_label : str
+            Label of the data to set the visiblity.  If not already loaded in the viewer, the
+            data will automatically be loaded before setting the visibility
+        visible : bool
+            Whether to set the layer(s) to visible.
+        replace : bool
+            Whether to disable the visilility of all other layers in the viewer
+        """
+        viewer_item = self._get_viewer_item(viewer_reference)
+        viewer_id = viewer_item['id']
+        viewer = self.get_viewer_by_id(viewer_id)
+
+        # if the data_label is in the app, but not loaded in the viewer, automatically load it first
+        viewer_data_labels = [layer.layer.label for layer in viewer.layers]
+        if data_label not in viewer_data_labels:
+            dc_labels = [data.label for data in self.data_collection]
+            if data_label not in dc_labels:
+                if os.path.exists(data_label):
+                    raise ValueError(f'The data label "{data_label}" is not available '
+                                     f'to add to the viewer, but it does specify a file path. '
+                                     f'If you intended to load the data from that file, use the '
+                                     f'`load_data` method or similar.')
+                raise ValueError(
+                    f"No data item found with label '{data_label}'. Label must be one "
+                    "of:\n\t" + "\n\t".join(dc_labels))
+
+            [data] = [x for x in self.data_collection if x.label == data_label]
+            color_cycler = ColorCycler()
+            viewer.add_data(data, percentile=95, color=color_cycler())
```

Comment:
To do/fix soon: this implementation always produce the first color in the cycler, for all data. I'll take a look!